src/utils.cpp

```diff
@@ -407,19 +407,33 @@ std::string resolve_binary_path(const std::string& cmd)
   }
 }
 
-void cat_file(const char *filename)
+void cat_file(const char *filename, size_t max_bytes)
 {
   std::ifstream file(filename);
-  std::string line;
+  const size_t BUFSIZE = 4096;
 
-  if (file.fail())
-  {
-    std::cerr << "ERROR: file not found: " << filename << std::endl;
+  if (!file){
```

Comment:
I think we should keep this as `file.fail()`. It took me a bit of searching to confirm they're the same, but the `fail` one is more obvious about what it's doing and it's what we use in the rest of the codebase